core/src/main/java/com/linecorp/armeria/server/ServiceConfig.java

```diff
@@ -123,6 +125,7 @@ private ServiceConfig(@Nullable VirtualHost virtualHost, Route route,
         this.multipartUploadsLocation = requireNonNull(multipartUploadsLocation, "multipartUploadsLocation");
         this.shutdownSupports = ImmutableList.copyOf(requireNonNull(shutdownSupports, "shutdownSupports"));
         this.defaultHeaders = defaultHeaders;
+        this.serviceErrorHandler = requireNonNull(serviceErrorHandler, "serviceErrorHandler");
```

Comment:
Not related to your PR, but let's just add `requireNonNull` to the `defaultHeaders` for consistency.
```suggestion
        this.defaultHeaders = requireNonNull(defaultHeaders, "defaultHeaders");
        this.serviceErrorHandler = requireNonNull(serviceErrorHandler, "serviceErrorHandler");
```